Quick/DSL/World+DSL.swift

```diff
@@ -17,8 +17,11 @@ extension World {
     }
 
     internal func describe(description: String, flags: FilterFlags, closure: () -> ()) {
+        guard currentExampleGroup != nil else {
+            fatalError("Error: example group was not created by its parent QuickSpec spec. Check that describe() was used in QuickSpec.spec() and not a more general context (i.e. an XCTestCase test)")
+        }
```

Comment:
Agreed, unit test is good since we're coupling this error message with assumptions that come from outside this class/file. I'm okay with using an `NSException` (and perhaps even writing a Swiftier wrapper around it) for the sake of testing in this case.